modules/core/src/smithy4s/schema/EnumTag.scala

```diff
@@ -16,28 +16,26 @@
 
 package smithy4s.schema
 
-sealed trait EnumTag[+E]
+// scalafmt: { maxColumn = 120, align.preset = most}
+sealed trait EnumTag[E]
 
 object EnumTag {
-  case object ClosedStringEnum extends EnumTag[Nothing]
-  case object ClosedIntEnum extends EnumTag[Nothing]
-  case class OpenStringEnum[E](unknown: String => E) extends EnumTag[E]
-  case class OpenIntEnum[E](unknown: Int => E) extends EnumTag[E]
+  case class StringEnum[E](value: E => String, unknown: Option[String => E]) extends EnumTag[E]
+  case class IntEnum[E](value: E => Int, unknown: Option[Int => E])          extends EnumTag[E]
```

Comment:
Would there be value now in also having `String => Option[E]` and `Int => Option[E]` at this level? Mostly just thinking of how we build the same-ish `Map[String, E]` and `Map[Int, E]` in the different interpreters

Reply:
Don't think so. The functions I added are not there for convenience, but rather because the removal of "total" makes it necessary for these to be present. 

It's easy enough to reconstruct maps on demand from the list of enum values.

Reply:
it kinda forces you to think about open enums explicitly because, when you do `values.groupBy(_.value).get(value)`, it's optional - meaning you have to consider kind of explicitly what happens if the value isn't there (which would be the case in open enums, as I understand)